bundles/es.elexis.server.core.connector.elexis.datasource/src/es/core/connector/elexis/datasource/internal/ElexisPoolingDataSource.java

```diff
@@ -0,0 +1,97 @@
+package es.core.connector.elexis.datasource.internal;
+
+import java.sql.Connection;
+import java.sql.Driver;
+import java.sql.SQLException;
+import java.util.Optional;
+import java.util.Properties;
+
+import javax.sql.DataSource;
+
+import org.apache.commons.dbcp.ConnectionFactory;
+import org.apache.commons.dbcp.DriverConnectionFactory;
+import org.apache.commons.dbcp.PoolableConnectionFactory;
+import org.apache.commons.dbcp.PoolingDataSource;
+import org.apache.commons.pool.ObjectPool;
+import org.apache.commons.pool.impl.GenericObjectPool;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.LoggerFactory;
+
+import ch.elexis.core.common.DBConnection;
+import es.core.connector.elexis.datasource.util.ElexisDBConnectionUtil;
+
+@Component(immediate = true, property = "osgi.jndi.service.name=jdbc/poolable")
+public class ElexisPoolingDataSource extends PoolingDataSource implements DataSource {
+	
+	@Activate
+	public void activate(){
+		Optional<DBConnection> dbConnection = ElexisDBConnectionUtil.getConnection();
+		if (dbConnection.isPresent()) {
+			ObjectPool<Connection> connectionPool = createConnectionPool(dbConnection.get());
+			if (connectionPool != null) {
+				setPool(connectionPool);
+				
+				try {
+					Connection conn = getConnection();
+					conn.close();
+					
+					LoggerFactory.getLogger(ElexisPoolingDataSource.class)
+						.info("db pool initialization success");
+				} catch (SQLException e) {
+					LoggerFactory.getLogger(ElexisPoolingDataSource.class)
+						.error("pool initialization error", e);
+				}
+			} else {
+				LoggerFactory.getLogger(ElexisPoolingDataSource.class)
+					.error("db pool initialization failed - no connection pool used");
+			}
+		}
+		
```

Comment:
We should somehow log the non-availability of a connection or something. How do bundles react on an inexistant connection?